src/Xrefcheck/Verify.hs

```diff
@@ -191,17 +191,19 @@ verifyRepo
   progressRef <- newIORef $ initVerifyProgress (map snd toScan)
 
   accumulated <- withAsync (printer progressRef) $ \_ ->
-    forConcurrentlyCaching toScan (\(_, Reference{..}) -> case locationType rLink of
-      ExternalLoc -> CacheUnderKey rLink
-      _           -> NoCaching) (\(file, ref) -> do
+    forConcurrentlyCaching toScan ifExternalThenCache $ \(file, ref) -> do
       VerifyResult results <- verifyReference config mode progressRef repoInfo' root file ref
-      return $ Left results)
-  return . foldMap VerifyResult $ lefts accumulated
+      return results
+  return . foldMap VerifyResult $ accumulated
```

Comment:
And last minor nitpick: wrapping/unwrapping `VerifyResult` now also looks redundant.